File: src/2025-02-23-deepclaude-try/cpp/exercises/practice/robot-name/robot_name.cpp
#include "robot_name.h"
#include <unordered_set>
#include <cstdlib>
#include <ctime>
#include <string>

namespace robot_name {

namespace {
    std::unordered_set<std::string> used_names;
    bool seeded = false;
}

robot::robot() {
    if (!seeded) {
        std::srand(std::time(nullptr));
        seeded = true;
    }
    generate_new_name();
}

std::string robot::name() const {
    return name_;
}

void robot::reset() {
    generate_new_name();
}

void robot::generate_new_name() {
    std::string new_name;
    do {
        // Generate two uppercase letters
        char letter1 = 'A' + std::rand() % 26;
        char letter2 = 'A' + std::rand() % 26;
        
        // Generate three digits
        int number = std::rand() % 1000;
        char digit1 = '0' + (number / 100) % 10;
        char digit2 = '0' + (number / 10) % 10;
        char digit3 = '0' + number % 10;
        
        new_name = {letter1, letter2, digit1, digit2, digit3};
    } while (used_names.count(new_name));

    // Remove old name from used_names if it exists
    if (!name_.empty()) {
        used_names.erase(name_);
    }

    name_ = new_name;
    used_names.insert(new_name);
}

}  // namespace robot_name

File: src/2025-03-16-08-42-59--a-deepclaude-full-python-0316-1/python/exercises/practice/robot-name/robot_name.py
import random
import string

class Robot:
    _used_names = set()
    
    def __init__(self):
        self._name = None
        self._generate_name()
    
    @property
    def name(self):
        return self._name
    
    def _generate_name(self):
        while True:
            # Generate a name with 2 uppercase letters followed by 3 digits
            letters = ''.join(random.choices(string.ascii_uppercase, k=2))
            digits = ''.join(random.choices(string.digits, k=3))
            new_name = letters + digits
            
            # Ensure the name is unique
            if new_name not in Robot._used_names:
                Robot._used_names.add(new_name)
                self._name = new_name
                break
    
    def reset(self):
        # Remove the current name from used names
        if self._name:
            Robot._used_names.remove(self._name)
        
        # Generate a new name
        self._generate_name()
